### ISSUE TITLE
Looks like the signature change broke something 

### ISSUE THREAD

**Issue by harshavardhana** [minio/minio-py#210]
~~~
$ python put_object.py 
Traceback (most recent call last):
  File "put_object.py", line 27, in <module>
    client.put_object('testbucket', 'my_key', file_stat.st_size, file_data)
  File "/home/harsha/repos/minio-py/minio/minio.py", line 400, in put_object
    self._stream_put_object(bucket, key, length, data, content_type)
  File "/home/harsha/repos/minio-py/minio/minio.py", line 585, in _stream_put_object
    upload_id = self._new_multipart_upload(bucket, key, content_type)
  File "/home/harsha/repos/minio-py/minio/minio.py", line 640, in _new_multipart_upload
    secret_key=self._secret_key)
  File "/home/harsha/repos/minio-py/minio/signer.py", line 83, in sign_v4
    if ignored_header in headers_to_sign:
UnboundLocalError: local variable 'headers_to_sign' referenced before assignment
~~~

**Comment by harshavardhana** [comment]
~~~
$ git diff
diff --git a/minio/signer.py b/minio/signer.py
index cd99e32..96e23ab 100644
--- a/minio/signer.py
+++ b/minio/signer.py
@@ -81,7 +81,7 @@ def sign_v4(method, url, headers=None, access_key=None, secret_key=None, content
 
     for ignored_header in ignored_headers:
         if ignored_header in headers_to_sign:
-            del headers_to_sign
+            del headers_to_sign[ignored_header]
 
     canonical_request, signed_headers = generate_canonical_request(method, parsed_url, headers_to_sign,
                                                                    content_hash_hex)
$ 
~~~

This works.. with that put_object.py works with Minio-SignatureV4

**Comment by fkautz** [comment]
Fixed with #211.
